backend/src/worker.rs

```diff
@@ -680,270 +1077,238 @@ print()
-                "finished deno code execution {}",
-                job.id
-            );
+            sqlx::query!(
+                "UPDATE script SET lock = $1 WHERE hash = $2 AND workspace_id = $3",
+                &content,
+                &job.script_hash.unwrap_or(ScriptHash(0)).0,
+                &job.workspace_id
+            )
+            .execute(db)
+            .await?;
+        }
+        Err(error) => {
+            sqlx::query!(
+                "UPDATE script SET lock_error_logs = $1 WHERE hash = $2 AND workspace_id = $3",
+                &format!("{logs}\n{error}"),
+                &job.script_hash.unwrap_or(ScriptHash(0)).0,
+                &job.workspace_id
+            )
+            .execute(db)
+            .await?;
         }
     }
     Ok(())
 }
 
-async fn handle_dependency_job(
-    job: &QueuedJob,
+async fn install_go_dependencies(
+    job_id: &Uuid,
+    code: &str,
     logs: &mut String,
     job_dir: &String,
     status: &mut Result<ExitStatus, Error>,
     db: &sqlx::Pool<sqlx::Postgres>,
     last_line: &mut String,
     timeout: i32,
-) -> Result<(), Error> {
-    let requirements = job
-        .raw_code
-        .as_ref()
-        .ok_or_else(|| Error::ExecutionErr("missing requirements".to_string()))?;
-    logs.push_str(&format!("content of requirements:\n{}\n", &requirements));
-    let file = "requirements.in";
-    write_file(job_dir, file, &requirements).await?;
-    let child = Command::new("pip-compile")
+    go_path: &str,
+    preview: bool,
+) -> error::Result<String> {
+    gen_go_mymod(code, job_dir).await?;
+    let child = Command::new("go")
         .current_dir(job_dir)
-        .args(vec!["-q", "--no-header", file])
+        .args(vec!["mod", "init", "mymod"])
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
         .spawn()?;
-    *status = handle_child(job, db, logs, last_line, timeout, child).await;
-    Ok(if status.is_ok() && status.as_ref().unwrap().success() {
-        let path_lock = format!("{}/requirements.txt", job_dir);
-        let mut file = File::open(path_lock).await?;
-
-        let mut content = "".to_string();
-        file.read_to_string(&mut content).await?;
-        content = content
-            .lines()
-            .filter(|x| !x.trim_start().starts_with('#'))
-            .map(|x| x.to_string())
-            .collect::<Vec<String>>()
-            .join("\n");
-        let as_json = json!(content);
+    *status = handle_child(job_id, db, logs, last_line, timeout, child).await;
+    if status.is_ok() {
+        let child = Command::new(go_path)
+            .current_dir(job_dir)
+            .args(vec!["mod", "tidy"])
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?;
+        *status = handle_child(job_id, db, logs, last_line, timeout, child).await;
+    }
+    if status.is_ok() && status.as_ref().unwrap().success() {
+        if preview {
+            Ok(String::new())
```

Comment:
if this is a preview, go.mod and go.sum is not written yet they are required bind mounts in run.go.config.proto

Reply:
They are written as a file, it's just that they are not returned as a string to attach as a lock file to the script.